cdap-docs/examples-manual/source/examples/log-analysis.rst

```diff
@@ -0,0 +1,189 @@
+The Spark program also computes the total number of requests made by every unique ip address and writes it to TimePartitionedFileSet, *reqCount*.
+
+In parallel, these events are also taken up by the *HitCounterProgram*, which goes through the entries, calculates the total number hits for every unique url and tabulates results in KeyValueTable dataset, *hitCount*.
+
+The *LogAnalysisWorkflow* ties the Spark and MapReduce to run in parallel.
+
+Once the application completes, you can query the *responseCount* dataset by using the ``rescount`` endpoint of the *ResponseCounterService*.
+It will send back a string result with total number of responses on the ``rescount`` query parameter. You can also query the
+*hitCount* dataset by using ``url`` endpoint of the *HitCounterService*. It will send the total number of hits for the queried url.
+You can query the ``reqCount`` TimePartitionedFileSet by using the ``reqcount`` endpoint of the *RequestCounterService* which will return a set of all the available partitions.
+Using one of partitions from the above set you can query for the total number of request made by every unique ip address in last 60 minutes by using the ``reqfile``
+endpoint of the *RequestCounterService* which will return a Map of ip addresses and total number of requests made by them.
+
+Let's look at some of these components, and then run the application and see the results.
+
+The LogAnalysis Application
+-----------------------------
+
+As in the other `examples <index.html>`__, the components
+of the application are tied together by the class ``LogAnalysisApp``:
+
+.. literalinclude:: /../../../cdap-examples/LogAnalysis/src/main/java/co/cask/cdap/examples/loganalysis/LogAnalysisApp.java
+   :language: java
+   :lines: 70-94
+
+The ``hitCount``, ``responseCount`` KeyValueTable and  ``reqCount`` TimePartitionedFileSet
+--------------------------------------------------------------------------------
+
+The calculated hit count for every unique url is stored in an KeyValueTable dataset, *hitCount* and the total number of responses for a response code is stored in KeyValueTable dataset, *responseCount*.
+The total number of requests made by every unique ip address is written to TimePartitionedFileSet, *ipCount*.
+
+The ``HitCounterService``, ``ResponseCounterService`` and ``RequestCounterService`` Services
+----------------------------
+
+This ``HitCounterService`` service has a ``hitcount`` endpoint to obtain the total number of hits for a given url.
+This ``ResponseCounterService`` service has a ``rescount`` endpoint to obtain the total number of responses for a given response code.
+This ``RequestCounterService`` service has a ``reqcount`` endpoint to obtain a set of all the available partitions in the TimePartitionedFileSet and it also
+has a ``reqfile`` to retrieve data from a particular partition.
```

Comment:
Let's re-do this into three bullets:
```
The ``ResponseCounterService`` service has a three endpoints:
- ``rescount`` endpoint to obtain the total number of responses for a given response code;
- ``reqcount`` endpoint to obtain a set of all the available partitions in the ``TimePartitionedFileSet``; and
- ``reqfile`` to retrieve data from a particular partition.
```

Do you think that the "rescount" and "reqcount" are too close in names? One obtains a count, while the other gives a set rather than a count. How about "reqcount" becoming "partitionmap"?